src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -74,7 +75,18 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>, g
     // At this point any memoization is no longer valid since we'll have changed
     // the children.
     workInProgress.memoizedProps = null;
-    if (current && current.child === workInProgress.child) {
+    if (!current) {
+      // If this is a fresh new component that hasn't been rendered yet, we
+      // won't update its child set by applying minimal side-effects. Instead,
+      // we will add them all to the child before it gets rendered. That means
+      // we can optimize this reconciliation pass by not tracking side-effects.
+      workInProgress.child = mountChildFibersInPlace(
```

Comment:
It looks like this change doesn't do anything currently because reconcileChildFibersInPlace and mountChildFibersInPlace are the same right now. Is that right?